src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -137,10 +152,20 @@ export const NavSearchBar: React.FC<NavSearchBarProps> = ({ venueId }) => {
     if (!searchQuery) return [];
 
     return relatedEvents
-      .filter((event) => event.name.toLowerCase().includes(searchQuery))
+      .filter((event) => {
+        const isEventRoomEnabled =
+          isDefined(event.room) && event.room !== ""
+            ? isDefined(enabledRelatedRoomsByTitle[event.room])
+            : true;
+
+        return (
+          isEventRoomEnabled && event.name.toLowerCase().includes(searchQuery)
+        );
+      })
       .map((event) => {
         const imageUrl =
-          relatedRooms.find((room) => room.title === event.room)?.image_url ??
+          enabledRelatedRooms.find((room) => room.title === event.room)
+            ?.image_url ??
```

Comment:
This should make use of `enabledRelatedRoomsByTitle[event.room]` rather than re-searching through them all again.